proof_ipa.go

```diff
@@ -404,6 +407,11 @@ func PreStateTreeFromProof(proof *Proof, rootC *Point) (VerkleNode, error) { //
 		poas  = proof.PoaStems
 	)
 
+	// The proof of absence stems must be sorted. If that isn't the case, the proof is invalid.
+	if !sort.IsSorted(bytesSlice(proof.PoaStems)) {
+		return nil, fmt.Errorf("proof of absence stems are not sorted")
+	}
```

Comment:
This is a check I pulled from the research repo. It's true that we must expect this slice to be sorted, so I do that here. I had to create a `bytesSlice` type (see bottom of this file).